fiftyone/core/sample.py

```diff
@@ -768,55 +765,53 @@ def _save(self, deferred=False):
         return sample_ops, frame_ops
 
 
-def _apply_confidence_thresh(label, confidence_thresh):
+def _apply_label_filters(label, confidence_thresh, classes):
+    if confidence_thresh is None and classes is None:
+        return label
+
     if _is_frames_dict(label):
         label = {
-            frame_number: _apply_confidence_thresh(
-                frame_dict, confidence_thresh
+            frame_number: _apply_label_filters(
+                frame_dict, confidence_thresh, classes
             )
             for frame_number, frame_dict in label.items()
         }
     elif isinstance(label, dict):
         label = {
-            k: _apply_confidence_thresh(v, confidence_thresh)
+            k: _apply_label_filters(v, confidence_thresh, classes)
             for k, v in label.items()
         }
+    elif isinstance(label, fol._HasLabelList):
```

Comment:
Note that `confidence_thresh` has different semantics for `Keypoint(s)` instances than other `_HasLabelList` types: the `confidence` attribute is an array of per-point confidences rather than a single value. That's why there was nearly duplicate code in the previous implementation,.

I'm sure there's a way to have slightly less duplicate code, but we will need to bring back the correct behavior for `Keypoint(s)` as this implementation will not work for them.